File: template-rest-assured-java/templates/src/test/java/{{package_name}}/support/utils/BaseRest.java
package{{package_name}}.support.utils;

import io.restassured.RestAssured;
import io.restassured.response.Response;
import io.restassured.specification.RequestSpecification;

import java.util.Map;

public class BaseRest {
    private static RequestSpecification request;
    private static Response response;

    public static void initRest(){
        RestAssured.useRelaxedHTTPSValidation();
        request = RestAssured.given();
    }

    public static void initRestWithProxy(){
        RestAssured.useRelaxedHTTPSValidation();
        RestAssured.proxy("proxy", 8080);
        request = RestAssured.given();
    }

    public static void log(){
        RestAssured.given().then().log().all();
    }

    public static void setPath(String baseUri, String path){
        if ( request == null){
            initRest();
        }
        request.baseUri(baseUri);
        request.basePath(path);
        RestAssured.useRelaxedHTTPSValidation();
    }

    public static void setContent(String content){
        if ( request == null){
            initRest();
        }
        request.contentType(content);
    }

    public static void setHeader(Map<String, String> contentHeader){
        request.headers(contentHeader);
    }

    public static void setPathParams(Map<String, String> params){
        request.pathParams(params);
    }

    public static void setParams(Map<String, ?> params){
        request.params(params);
    }

    public static void setQueryParams(Map<String, ?> queryParams){
        request.queryParams(queryParams);
    }

    public static void setFormParams(Map<String, ?> formParams){
        request.formParams(formParams);
    }

    public static void setBody(String contentBody){
        request.body(contentBody);
    }

    public static void getRequest(){
        response = request.get();
    }

    public static void postRequest(){
        response = request.post();
    }

    public static void deleteRequest(){
        response = request.delete();
    }

    public static void putRequest(){
        response = request.put();
    }

    public static void patchRequest(){
        response = request.patch();
    }

    public static Response getResponse(){
        return response;
    }

}

File: template-rest-assured-kotlin/templates/src/test/kotlin/{{package_name}}/support/utils/BaseRest.kt
package {{package_name}}.support.utils

import io.restassured.specification.RequestSpecification
import io.restassured.RestAssured
import io.restassured.response.Response

object BaseRest {
    private var request: RequestSpecification? = null
    var response: Response? = null


    fun initRest() {
        RestAssured.useRelaxedHTTPSValidation()
        request = RestAssured.given()
    }

    fun initRestWithProxy() {
        RestAssured.useRelaxedHTTPSValidation()
        RestAssured.proxy("proxy", 8080)
        request = RestAssured.given()
    }

    fun log() {
        RestAssured.given().then().log().all()
    }

    fun setPath(baseUri: String?, path: String?) {
        if (request == null) {
            initRest()
        }
        request!!.baseUri(baseUri)
        request!!.basePath(path)
        RestAssured.useRelaxedHTTPSValidation()
    }

    fun setContent(content: String?) {
        if (request == null) {
            initRest()
        }
        request!!.contentType(content)
    }

    fun setHeader(contentHeader: MutableMap<String, String>) {
        request!!.headers(contentHeader)
    }

    fun setPathParams(params: Map<String?, String?>?) {
        request!!.pathParams(params)
    }

    fun setParams(params: Map<String?, *>?) {
        request!!.params(params)
    }

    fun setQueryParams(queryParams: Map<String?, *>?) {
        request!!.queryParams(queryParams)
    }

    fun setFormParams(formParams: Map<String?, *>?) {
        request!!.formParams(formParams)
    }

    fun setBody(contentBody: String?) {
        request!!.body(contentBody)
    }

    fun getRequest() {
        response = request!!.get()
    }

    fun postRequest() {
        response = request!!.post()
    }

    fun deleteRequest() {
        response = request!!.delete()
    }

    fun putRequest() {
        response = request!!.put()
    }

    fun patchRequest() {
        response = request!!.patch()
    }

}